### DISCUSSION TITLE
Quarto render and preview inconsistently rendering draft...

### DISCUSSION THREAD

**Discussion by astrowonk** [quarto-dev/quarto-cli#10239] [Q&A]
### Description

I wish I had an easy reproduction. My web site is in quarto. I created a new post inside my structure. if `draft:true` is on the post, the post completely fails to render. This is currently a text-only post with no executable code at all. The resulting html file is:

```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"></html>
```
and nothing else.

if `draft:false` it renders and previews fine. I made a standalone `.qmd` file without the structure of my site, and it worked without issue, both in draft and not in draft. But within my site setup, the draft just will not render.

Is there something I'm doing wrong? Logging did not show any obvious errors etc.

**Comment by astrowonk** [comment]
The issue is also present in the 1.6.x preview but the drafts render fine in 1.4.557 (which doesn't seem to show the Draft banner at the top, as I think that was a 1.5.x feature?)

The issue isn't present in 1.5.0 but does occur in 1.5.22. I think I'm just going back to 1.4.x for now.

**Reply by astrowonk** [reply]
I have a [repo](https://github.com/astrowonk/quarto-draft-bug) that reproduces the issue along with a screen recording. This is on Mac OS Sonoma 14.5 installed with the regular quarto 1.5 installer on the quarto.org web site. Happy to open an issue if that's better than here.


https://github.com/quarto-dev/quarto-cli/assets/13702392/98054ba7-4ffa-4b64-914f-5f2fef35cdad

**Reply by mcanouil** [reply]
I am not sure the issue you are showing is about the draft mode as clearly the preview showed the draft (at least at first).

Are the other pages preview update on changes in your setup?
Does only a draft page preview break?

**Reply by mcanouil** [reply]
To be clear, is you screen recording about the same issue as your original post or something else?
Because you stated that Quarto does not preview draft pages but your recording shows that it does.
Additionally, you are comparing behaviour with versions that does not even have/support draft, so it's a bit confusing.

**Comment by mcanouil** [comment]
It does not work for versions earlier than 1.5 (1.5.0 did not have the feature) because the draft feature had no meaning at that time.

The purpose of a draft is not to be rendered when rendering a project, so I am not sure why you would expect a draft to be rendered as if it was not a draft.

How are you calling `quarto preview`? If you are calling it on the project or on any files that is not in draft mode, then the expected behaviour is to not render draft as long as you don't navigate to it (same applies for other pages).

Did you read the documentation of the feature you are trying to use?

https://quarto.org/docs/websites/website-drafts.html

**Reply by astrowonk** [reply]
😔 I swore I did read up on drafts, but _clearly_ I missed this very obvious explanation in the documentation. 🤦🏽‍♂️ In 1.4, 'draft' essentially means 'unlinked' in 1.5, right? I clearly did not read that documentation page carefully, because I thought that a draft just meant "not rendered on the listing page" as in 1.4. The fact I sometimes saw a rendered draft with "Draft" at the top reinforced that belief.

But, as you can see in my video even though I don't have any setting for `draft-mode: unlinked` anywhere in my example repo, the draft *did* render as if set to unlinked. I think that's what confused me.

In the video I call quarto preview via the quarto plugin. And yes, the draft is there… but then it _isn't_ as soon as I edit it. So, I think what I have found is a _different_ bug than I originally thought. It looks like the behavior is shifting from `draft-mode: unlinked` to `draft-mode: gone` in some unpredictable way. Because in the video I don't change any yml settings. I just edit the file and the draft goes from rendered to not, both done with quarto preview via the VS code plugin.

**Reply by astrowonk** [reply]
Wait, perhaps the first invocation of `preview` is on just the draft file.. and the subsequent ones (which occur automatically when I edit) are on the entire project thus the draft disappears. So not so much a bug, but a confusing interaction between preview and watch-inputs.

**Reply by mcanouil** [reply]
I need to check myself the behaviour now to investigate the possible preview bug.
I'll come back to you.

(I'am unmarking the thread for now)

**Reply by mcanouil** [reply]
I've opened an issue to track the preview issue with draft on modification.

- #10279
